.github/workflows/javadocs_scripts/libraryTable_generation.py

```diff
@@ -0,0 +1,80 @@
+# This script recreates the table of modules for the site
+
+import yaml
+
+# Load the yaml files
+with open('site/data/variables.yaml', 'r') as f:
+  data = yaml.safe_load(f)
+
+with open('site/data/variableLibraryReferenceLinks.yaml', 'r') as f:
+  libraryReference = yaml.safe_load(f)
+
+with open('site/data/variableProductReferenceLinks.yaml', 'r') as f:
+  productReference = yaml.safe_load(f)
+
+with open('site/data/name_pretty.yaml', 'r') as f:
+  nameReference = yaml.safe_load(f)
+
+
+# List of modules to ignore
+variables_to_skip = ['libraries-bom', 'gapic-libraries-bom','github-repo','gax-httpjson', 'google-cloud-shared-dependencies', 'first-party-dependencies']
+
+# List of runtime modules
+runtime_modules = ['google-cloud-core', 'gax', 'google-iam-policy', 'api-common']
+
+# List of modules with special artifact names
+variable_name_exceptions = ['google-cloud-bigquerystorage-bom', 'google-cloud-bigtable-bom','google-cloud-datastore-bom', 'google-cloud-firestore-bom','google-cloud-logging-bom', 'google-cloud-pubsub-bom', 'google-cloud-pubsublite-bom', 'google-cloud-spanner-bom', 'google-cloud-storage-bom']
```

Comment:
I hear your concern - let me address for each hardcoded list:

1) For modules to ignore - I've updated the scripts such that the only to places where this harcoded list exists are in the `update_googlecloudjava_javadocs_modules.py` and `libraryTable_generation.py`. 

`update_googlecloudjava_javadocs_modules.py` needs to have the list as we are generating aggregated Javadocs for only the modules that start with `google-cloud-*` and not individually for the `proto-google`/`grpc-google` ones.

`libraryTable_generation.py` needs the list to guard against adding any artifacts to the site that should not be included, even if they're included in the `site/data/javaModulesVersions.yaml` file.

2) For runtime vs. product categorization, the only thing I can think of is that the runtime modules are within the sdk-platform-java repo and all others are product. Since I'm already manually hardcoding the modules we're generating the javadocs for in the `sdk-platform-java_javadocs_modules.txt` files, I've refactored this script to get the list of runtime modules from that file, so at least if we need to update it, then we only need to update in that file.

3) For variable_name_exceptions - these are limited to the handwritten libraries so my sense is that list does not change often. I hear you that it's clunky; the automation for these scripts is coming from each repo's .repo-metadata.json file, so if the handwritten libraries were able to update the `distribution_name` in their files to have the `-bom` suffix, the `variable_name_exceptions` could get removed (e.g. https://github.com/googleapis/java-bigtable/blob/main/.repo-metadata.json#L11C5-L11C5) 

I'm not sure if the repos are using that file/field for anything, so I'll bring it up and see what they say.

Reply:
Okay based on some insight from the team, it seems like `distribution_name` in the `.repo-metadata.json` files *used* to be used for the publish_javadoc jobs and that spring-cloud-gcp is using them to get the group ID/artifact names (only the google-cloud-java modules for now). 
So we *could* propose that each handwritten repo update that field to include the `-bom` suffix; however, I'm not sure this specific use case is compelling enough to warrant that? It seems like less hassle for us to maintain this list of exceptions here than push a bunch of PRs to each handwritten repo; but happy to discuss further.